src/Ethernet.cpp

```diff
@@ -224,6 +231,32 @@ void EthernetClass::setRetransmissionCount(uint8_t num)
 	SPI.endTransaction();
 }
 
+void EthernetClass::generateDefaultHostName(uint8_t *mac) {
+	// Copy the default host name base
+	strcpy(_hostName, DEFAULT_HOST_NAME);
+
+	// Append the last 3 bytes of the MAC (HEX'd)
+    char macAddrStr[3];
+    sprintf(macAddrStr, "%02X", mac[3]);
+    strcat(_hostName, macAddrStr);
+    sprintf(macAddrStr, "%02X", mac[4]);
+    strcat(_hostName, macAddrStr);
+    sprintf(macAddrStr, "%02X", mac[5]);
+    strcat(_hostName, macAddrStr);
```

Comment:
Could this be reduced to one line?
`sprintf_P(_hostName, PSTR("%02X%02X02X"), mac[3], mac[4], mac[5])`

`PSTR()` saves the format string in program memory (Flash) instead of RAM. I haven't tested my suggestion, just thought it may help shrink code size.

Reply:
Good idea. I changed a bit on your suggestion to include the default hostname as well.
Works nicely on my setup. I don't know, how memory usage can be tested. the compiler gives the same information in either case:
"Sketch uses 12230 bytes (39%) of program storage space. Maximum is 30720 bytes.
Global variables use 714 bytes of dynamic memory."

Reply:
The report from the Github bot shows change  in arduino:avr:mega:

Before:
* Flash memory: +1574 bytes
* SRAM: +28 bytes

After:
* Flash memory: +1482 bytes
* SRAM: +22 bytes

A very welcome change :) 

Now I wonder what is adding over 1KB of memory? 🤔 It seems like a lot.

Reply:
Could the 1KB increase be explained by the compiler deciding to inline some functions on its own? I've seen that before when LTO is enabled (I'm not sure if that's the case in the default Arduino builds).

Reply:
I refactored this again. This time I removed the usage of `sprintf`, and copied the MAC part on a bit lower level.
Now, I can see much smaller numbers on the "Memory usage change".
Can you check this, please?

Reply:
It looks correct to me. 👍